### PR TITLE
registration: Change occupation to dropdown list

### PR DESCRIPTION
**Pull request by tcluri**
Change occupation to a dropdown list from text

### CODE REVIEW

File: frontend/src/constants.js [hunk 1]
```diff
@@ -48,6 +48,16 @@ export const stateChoices = [
   { value: "WB", label: "West Bengal" }
 ];
 
+export const occupationChoices = [
+  {value: "Student - College/University(above 18 years old)", label: "Student - College/University(above 18 years old)"},
```
Comment (punchagan) [inline frontend/src/constants.js:52]: I think we can remove the "above 18 years old" part. Even the College/University doesn't make much sense, but maybe leave it in, for now.

File: frontend/src/constants.js [hunk 2]
```diff
@@ -48,6 +48,16 @@ export const stateChoices = [
...
+  {value: "Working - Own business", label: "Working - Own business"},
```
Comment (punchagan) [inline frontend/src/constants.js:53]: I think the "Working - " prefixes don't add much.

File: frontend/src/constants.js [hunk 3]
```diff
@@ -48,6 +48,16 @@ export const stateChoices = [
...
+  {value: "Working - Government job", label: "Working - Government job"},
+  {value: "Working - NGO/NPO", label: "Working - NGO/NPO"},
+  {value: "Working - Other", label: "Working - Other"},
+  {value: "Not working/Unemployed", label: "Not working/Unemployed"}
```
Comment (punchagan) [inline frontend/src/constants.js:57]: Just use unemployed.

**Review by punchagan — CHANGES_REQUESTED** [review]
Thanks for working on this.  I had minor comments on the field options. Can you address them? 

I think we could also add validation on the backend, using `choices` on the model, so that the imported data is in sync with the data coming in from the form.

File: server/models.py [hunk 4]
```diff
@@ -69,7 +69,16 @@ class StatesUTs(models.TextChoices):
     state_ut = models.CharField(max_length=5, choices=StatesUTs.choices, null=True, blank=True)
     not_in_india = models.BooleanField(default=False)
     team_name = models.CharField(max_length=100)
-    occupation = models.CharField(max_length=100, null=True, blank=True)
+
+    class OccupationTypes(models.TextChoices):
+        STUDENT = "Student", _("Student")
+        BUSINESS = "Business", _("Own business")
+        GOVERNMENT = "Government", _("Government")
+        ORGS = "Organizations", _("NGO / NPO")
```
Comment (punchagan) [inline server/models.py:77]: `Organisations` is a weird value, no?

File: server/management/commands/import_data.py [hunk 5]
```diff
@@ -191,6 +206,15 @@ def handle(self, *args, **options):
                     not_in_india = True
                     state_ut = None
 
+                if not minors:
+                    occupation = clean_occupation(row[columns["occupation"]])
+                    if occupation in OCCUPATIONS:
```
Comment (punchagan) [inline server/management/commands/import_data.py:211]: This check could go into the `clean_occupations` function to simplify the code here.

File: server/management/commands/import_data.py [hunk 6]
```diff
@@ -108,6 +109,20 @@ def clean_phone(phone):
         return ""
 
 
+def clean_occupation(occupation):
+    if occupation is None:
+        return None
+    if occupation.startswith("Student"):
```
Comment (punchagan) [inline server/management/commands/import_data.py:115]: Can be an `elif`.

**Review by punchagan — COMMENTED** [review]
I'm probably being a little nit-picky here, but some small changes would make this code cleaner.

**Comment by punchagan** [comment]
I've merged some conflicting changes on `main`.  Could you rebase against latest `main`. You'll need to recreate your migrations file too, since I've added another migration with the number `0009`.

File: server/models.py (hunk 4 above)
Comment (tcluri) [reply server/models.py]: Changed to "Non-profit"

File: server/management/commands/import_data.py (hunk 5 above)
Comment (tcluri) [reply server/management/commands/import_data.py]: moved it

**Comment by punchagan** [comment]
Merged manually with a couple of changes to `clean_occupation` to make it less verbose and more readable.  Thanks!

**Comment by punchagan** [comment]
Closes #15

**Pull request closed without merging**